units/src/weight.rs

```diff
@@ -198,6 +198,20 @@ crate::internal_macros::impl_op_for_references! {
 
         fn div(self, rhs: Weight) -> Self::Output { self.to_wu() / rhs.to_wu() }
     }
+    impl ops::Rem<u64> for Weight {
+        type Output = Weight;
+
+        fn rem(self, rhs: u64) -> Self::Output {
+            Weight(self.0 % rhs)
+        }
+    }
+    impl ops::Rem<Weight> for Weight {
+        type Output = Weight;
+
+        fn rem(self, rhs: Weight) -> Self::Output {
+            Weight(self.0 % rhs.0)
+        }
+    }
```

Comment:
We need these for references as well. I didn't look but we should be able to use the new macro `impl_op_for_references`.

Reply:
> The impl is already using the macro

My bad.

> is there are reason for this

Both used to be correct but as of #4090 we should not be usinging `.0` anywhere. I'll ack this as is and you could do another PR that enforces the new policy in `weight` if you feel like it?